django/contrib/gis/db/backends/mysql/operations.py

```diff
@@ -54,13 +63,21 @@ def gis_operators(self):
             operators["relate"] = SpatialOperator(func="ST_Relate")
         return operators
 
-    disallowed_aggregates = (
-        models.Collect,
-        models.Extent,
-        models.Extent3D,
-        models.MakeLine,
-        models.Union,
-    )
+    @cached_property
+    def disallowed_aggregates(self):
+        disallowed_aggregates = [
+            models.Extent,
+            models.Extent3D,
+            models.MakeLine,
+            models.Union,
+        ]
+        if self.connection.mysql_is_mariadb or self.connection.mysql_version < (
+            8,
+            0,
+            24,
+        ):
```

Comment:
Maybe we can add `is_mariadb` variable to avoid this wrapping?
```suggestion
        is_mariadb = self.connection.mysql_is_mariadb
        if is_mariadb or self.connection.mysql_version < (8, 0, 24):
```